### PR TITLE
Added option to display Surah Header and Skip BismiLlah

### PR DESCRIPTION
**Pull request by samwids**
1) Added a CLI option to display the Surah header, including its font size and top margin, in generated video, (below top margin for video in portrait mode 720x1280)  e.g.:

CLI =  --show-surah-header --surah-header-font-size 40 --surah-header-margin-top 200
config.json = 
  "surahHeaderFont": {
    "family": "Alvi Nastaleeq",
    "file": "fonts/Alvi Nastaleeq Regular.ttf",
    "size": 50,
    "color": "FFFFFF"
  },

2) Added a CLI option to skip Bismillah. When skipping it, the Ayah number was missing from the first Ayah, so this was handled separately:

CLI = --skip-start-bismillah

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -15,16 +15,22 @@
   "arabicFont": {
     "family": "KFGQPC HAFS Uthmanic Script",
     "file": "fonts/UthmanicHafs_V22.ttf",
-    "size": 80,
+    "size": 70,
     "color": "FFFFFF"
   },
   "translationFont": {
```
Comment (ashaltu) [inline config.json]: Likewise keep English translation font as the default config.json values?

File: src/config_loader.cpp [hunk 2]
```diff
@@ -260,6 +260,17 @@ AppConfig loadConfig(const std::string& path, CLIOptions& options) {
     cfg.arabicFont.size = data["arabicFont"].value("size", 100);
     cfg.arabicFont.color = data["arabicFont"].value("color", "FFFFFF");
 
+	// Load surahHeaderFont configuration  
+	if (data.contains("surahHeaderFont") && data["surahHeaderFont"].is_object()) {  
+		cfg.surahHeaderFont.family = data["surahHeaderFont"].value("family", cfg.arabicFont.family);
```
Comment (ashaltu) [inline src/config_loader.cpp]: Using the default values in the config you provided shows an Urdu header:

Image

But the fall back you wrote is to use the Arabic font for the Surah header, but it doesn't seem to be rendering properly:

Image

Could you address this?

File: src/subtitle_builder.cpp [hunk 3]
```diff
@@ -172,12 +172,44 @@ std::string buildAssFile(const AppConfig& config,
...
+		  
+		// Use Arabic font for the header (no font fallback needed)  
+		std::string header_text_render = "{\\fn" + config.surahHeaderFont.family + "}" + header_text;  
+	  
+		// Start AFTER intro duration to avoid duplicate display  
+		double header_start_time = intro_duration + pause_after_intro_duration;  
+	  
+		ass_file << "Dialogue: 0," << format_time_ass(header_start_time) << ","  
+				<< format_time_ass(total_video_duration)  
+				<< ",Translation,,0,0,0,,{\\an8\\pos(" << config.width/2 << "," << header_y_position << ")"
```
Comment (ashaltu) [inline src/subtitle_builder.cpp]: You are using `Translation` here, so the final video shows the Translation font, this probably the source of the bug mentioned in the other comment. Did you mean to use the `Arabic` font, especially since there's a bunch of logic above where you write out the Surah in Arabic.

Alternatively, if you really do want the surah in the translation font, you can utilize the jsons in `data/misc` and `data/surah-names`. The util functions to access that stuff is in `localization_utils.cpp`.

**Review by ashaltu — CHANGES_REQUESTED** [review]
Assalamu alaykum,

JazzakAllahu Khair for the changes and your patience, things are a bit untidy. Some other notes:

1. Update the README with the new CLI flags.
2. Include an [Unreleased] section in CHANGELOG.md. Here is an example commit (it doesn't have to be verbose like this, short and sweet is fine): https://github.com/ashaltu/quran-video-maker-ffmpeg/commit/6cb9df186c7f9a8ca367df77ac5280bc5d6d4fd5
3. Since there isn't complete testing setup, can you list out the different commands you ran to test edge cases. Screenshots are helpful too and save a bunch of time reviewing, otherwise I'll just manually re-run each command to verify.

File: src/config_loader.cpp (hunk 2 above)
Comment (samwids) [reply src/config_loader.cpp]: 13. Removed the 'surahHeaderFont' section that was previously added in 'config.json'. The project now uses consistent Arabic Surah names as headers.
14. Added 'ar.json' in 'data/surah-names' for Surah headers. This fixes issues where Surah names were not rendering correctly due to text/font incompatibility.
260308-171456-vlc-Surah_-_2_182_183_-_Al-Baqarah_-_t4_r25 mp4_-_VLC_-E5kX4

File: src/recitation_utils.cpp [hunk 4]
```diff
@@ -21,7 +21,15 @@ void normalizeGaplessTimings(std::vector<VerseData>& verses) {
 
 VerseData buildBismillahFromTiming(const TimingEntry& timing,
                                    const AppConfig& config,
-                                   const std::string& localAudioPath) {
+                                   const std::string& localAudioPath, 
+								   bool skipStartBismillah) {
+
+	if (skipStartBismillah) {
```
[the remark this answers is not in the document]
Comment (samwids) [reply src/recitation_utils.cpp:27]: 17. Fixed an issue where two Ayah numbers were being displayed.
260308-154407-vlc-Surah_-_1_1_7_-_Al-Fatiha_-_t4_r25 mp4_-_VLC_media-R5PEZ

**Comment by samwids** [comment]
Salaam,

I sincerely apologize for getting back so late, as I did not get the time to work on this or properly follow your advice. Unfortunately, it is not feasible for me due to my extremely busy schedule, and I cannot provide any commitment at this time.

I used and modified this project for my specific requirements, but the changes may not meet your well-defined coding standards or documentation expectations. Also, I am not very well-versed in C++ anymore, as it has been over 15–20 years and many things have changed since then.

However, I have committed some of my enhancements and listed the functions and changes I added below. If you find any of them useful when comparing with your commits, please feel free to incorporate them in whatever way you think is best.

Also, feel free to let me know if I can help in any way. As mentioned, however, I may not be able to provide any firm commitment due to my current engagements.

Changes and additions:

1. Added French translation.
2. Updates related to the new translation:
   * Updated 'quran_data.h' for French translation (id = 5).
   * Updated 'numbers.json' in 'data/misc'.
   * Updated 'surah.json' in 'data/misc'.
   * Added translation file 'fr-hamidullah-clean.json' in 'data/translations/fr/'. Translation got from tanzil.net
3. Added CLI parameter '--show-surah-header'.
4. Added CLI parameter '--surah-header-font-size'.
5. Added CLI parameter '--surah-header-margin-top'.
6. Added CLI parameter '--skip-start-bismillah'.
7. Added CLI parameter '--translation-font-size'.
8. Added CLI parameter '--translation-font-color'.
9. Updated 'ur-fatah-muhammad-jalandhari-simple.json' in 'data/translations/urd' to fix brackets and exclamation marks so they render correctly with the Urdu font.
10. Changed the default/fallback font from 'American Captain.ttf' to 'Roboto-Regular.ttf'.
11. Added 'Roboto-Regular.ttf' in 'assets/fonts' to be used as a common font for French, English, and other LTR languages.
12. Removed 'family' and 'size' from the 'translationFont' section in 'config.json'. These are now read from 'quran_data.h' depending on whether the translation language is LTR or RTL.
13. Removed the 'surahHeaderFont' section that was previously added in 'config.json'. The project now uses consistent Arabic Surah names as headers.
14. Added 'ar.json' in 'data/surah-names' for Surah headers. This fixes issues where Surah names were not rendering correctly due to text/font incompatibility.
15. Set the default video dimension and quality to '720x1280' (portrait), which was required for my use case (social media).
16. Changed the output MP4 filename format to:
    'Surah - [SurahNumber]_[StartAyat]_[EndAyat] - [Surah Name] - t[TranslationId]_r[ReciterId].mp4'
	Example: 'Surah - 44_10_10 - Ad-Dukhan - t4_r25.mp4'
17. Fixed an issue where two Ayah numbers were being displayed.
18. Updated data.tar for new french translation and fixed Arabic surah names, compressed as data.zip


Some of the changes are specific to my own requirements and may not be suitable for the main project (for example, portrait video mode or the output filename format I use for my automation). Please feel free to ignore those if they are not relevant.

e.g. CLI

Urdu
./build/qvm.exe 44 10 13 --reciter 25 --translation 4 --translation-font-size 55 --translation-font-color FFFFFF --show-surah-header --surah-header-font-size 50 --surah-header-margin-top 100 --skip-start-bismillah

./build/qvm.exe 1 1 7 --reciter 25 --translation 4 --translation-font-size 55 --translation-font-color FFFFFF --show-surah-header --surah-header-font-size 50 --surah-header-margin-top 100 --skip-start-bismillah


French
./build/qvm.exe 44 10 13 --reciter 25 --translation 5 --translation-font-size 35 --translation-font-color FFFFFF --show-surah-header --surah-header-font-size 50 --surah-header-margin-top 100 --skip-start-bismillah
